### DISCUSSION TITLE
'Waiting for server' time during calls to Swell

### DISCUSSION THREAD

**Discussion by Kealtie** [swellstores/community#196] [Help] [answered]
There's a lot of fluctuation in the speed which the server responds to the calls.

In the screenshot I was fetching products with a couple of filters applied. Sometimes this takes 200ms, sometimes I have to wait up to 3-4 seconds. How come?

<img width="2752" alt="image" src="https://user-images.githubusercontent.com/7847482/229751041-7f08e21d-74c1-4173-b112-642a20022eef.png">

**Comment by Kealtie** [comment]
Payload: 
<img width="432" alt="image" src="https://user-images.githubusercontent.com/7847482/229751181-66462a3f-9e5d-414f-be78-de93faffcb83.png">

**Reply by ericingram** [reply, marked as answer]
We believe this is due to API infrastructure spread across multiple providers, which we are very close to solving by moving to a single cloud. After that we'll be digging deeper on overall API performance improvements.

**Comment by fxlelouarn** [comment]
Thanks @ericingram . Do you have any idea as when this move will be completed? We're having the same problem (our calls are coming from Europe).

I've sent this to Swell's support but haven't received any answer. We still can change our underlying e-commerce solution without breaking all that we've built already. But, as we're starting developing features involving coupons, stock, etc., we're really at a crossroads : should we continue with Swell or should we try something else?

Thanks.

<img width="1334" alt="Capture d’écran 2023-05-11 à 16 33 33" src="https://github.com/swellstores/community/assets/115778739/e511995b-b259-44a9-97b6-649141f1bbcf">

**Reply by ericingram** [reply]
@fxlelouarn If your server is outside the US, that could explain the response times. We benchmarked an increase of ~1.5s for API calls in Europe/Asia. There are plans to add data centers in those regions in the future.

**Reply by fxlelouarn** [reply]
Thanks @ericingram! That would explain it, I guess.

**Reply by leonkosterss** [reply]
@ericingram Can you tell more about what the future is? Can we expect some changes in August? We want to go live with a new shop next month, but it is most of the time too slow (1-3 seconds). That is crucially for category and product pages..

**Reply by ericingram** [reply]
@leonkosterss we have several initiatives in progress still including improved server-side and client caching, and an EU-based API/database cluster, but in the meantime would recommend a specialized caching approach or static builds to avoid API calls as much as possible.

**Reply by christiancode14** [reply]
Any update on this? @ericingram

**Reply by ericingram** [reply]
@christiancode14 reach out to our support team and ask as we might have capacity to test the EU cluster

There is a major upgrade that is in progress with our master database that might still limit performance improvement, but our goal is to enable multi-region store support for all users eventually. Would be glad to have your input and feedback in the process.

**Comment by leonkosterss** [comment]
@ericingram We got answer from the support team it was not available yet (october 17). 

"Hi Leon,
I wanted to get back to you regarding your question about the beta testing of our EU server functionality.
We have discussed this with our infrastructure team, and they have informed us that the functionality of our EU server cluster is currently under development and is still being prepared for external testing. At the project's current state, we have limited information to answer your question if this is something you can switch easily. However, once this feature is ready, we plan to invite members of our user base to participate in testing the functionality and measuring the response time.
For now, we can add you to the list of users who have requested access for testing so that you will be notified once it becomes available.
Best,
Jesse B.
Swell
Customer Support"

**Can we get an invite already? Our webshop is too slow for now, and we can't go live with this poor performance..**

This call is taking from 1.92s to 11 seconds...

<img width="1064" alt="image" src="https://github.com/swellstores/community/assets/51200981/500d2c27-0799-4967-bc13-78734a7da8a5">


https://obimex-bv.swell.store/api/products?expand=categories&expand=variants&expand=attributes&categories=63ce9ed0d9bf6f001101a554&categories=65429fbeb7162200115fe0bc&categories=65429fbe1a3c1b0012d65aa1&categories=65429fc0da4a3900129f9628&categories=65429fbf8e821f00128d69a9&categories=65429fba0490d100120ec837&categories=65429fbb0490d100120ec839&categories=65429f9fe1a59e00125dfb51&categories=65429fb201c3a80013b88f27&categories=65429fb14f05750012f7b551&categories=65429fb0a50bdd0012b4c73f&categories=65429fb010e600001174f733&categories=65429faf4f05750012f7b54f&categories=65429faee68a420013654b63&categories=65429fa2da4a3900129f9625&categories=65429fa0b7162200115fe0b9&categories=65429f9ff315600012d445ae&categories=65429f97e1a59e00125dfb4f&categories=6542a003562d250012c675e1&categories=65429f9e6937b4001203237f&categories=65429f9da5f9040012f588d7&categories=65429f9a4f05750012f7b54d&categories=65429f99b7162200115fe0b4&categories=65429f99a5f9040012f588d5&categories=65429f98b5553b0011b4e02b&categories=65429f97b55ca30012ccf974&categories=65429f93a2d9830012e116fe&categories=6542a0056937b40012032386&categories=6542a004b5553b0011b4e02d&categories=6542a0028e821f00128d69ae&categories=65429fbcf315600012d445b2&categories=65429f93b87630001365c3dd&categories=65429f91dd5cfc0012351884&categories=65429f8d1a3c1b0012d65a98&categories=65429f8e92c02b001263435f&categories=65429f796937b40012032379&categories=65429f7a562d250012c675cd&categories=65429f6aa2d9830012e116fc&categories=6542a00351fd890012085960&categories=65429f96a2d9830012e11700&categories=65429f957d398d0012da5498&categories=65429f950490d100120ec832&categories=65429f948e5a8400121bb1bb&categories=65429f92605cd200124c8801&categories=65429f91e1a59e00125dfb4d&categories=65429f90e6c3d70012b65c3b&categories=65429f8ff315600012d445a9&categories=65429f8e42be9c001261f74c&categories=65429f890490d100120ec830&categories=65429f82da4a3900129f9623&categories=65429f80a5f9040012f588ce&categories=65429f8042be9c001261f74a&categories=65429f7f6937b4001203237c&categories=65429f7e6a1d5600128c7474&categories=65429f7e605cd200124c87ff&categories=65429f7d4e797d0012e4acd8&categories=65429f7c5aec500012b13d45&categories=65429f7c01c3a80013b88f13&categories=65429f7b10e600001174f731&categories=65429f7a355d2b00125fdee1&categories=64af9ee80d8c0c001226b143&categories=63ce9ee2d9bf6f001101a55b&limit=100&page=1

**Reply by ericingram** [reply]
Have you considered caching or static generation for primary pages?

At this point we have some more major changes to make before EU becomes available. Can't say exactly when, but you'll be on the list as soon as its ready.

**Comment by leonkosterss** [comment]
Any updates on EU servers? Updating a cart still takes 1 second.
new.obimex.nl

```
https://obimex-bv.swell.store/api/cart/items/667d2a362489830012b2cf7a
Request Method:
PUT
```

<img width="1082" alt="image" src="https://github.com/swellstores/community/assets/51200981/e2c0c235-3ed5-43b1-a20a-b47ed841dc68">

**Comment by genox** [comment]
Chiming in to add that we are also looking forward to seeing lower roundtrip times on requests from the EU.

**Comment by leonkosterss** [comment]
Better switch to MedusaJS. Op 8 mei 2025 om 23:09 heeft Oliver Studer ***@***.***> het volgende geschreven:﻿
Chiming in to add that we are also looking forward to seeing lower roundtrip times on requests from the EU.
